iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -690,15 +690,11 @@ ParseResult ExecutableEntryPointOp::parse(OpAsmParser &parser,
   }
   result.addAttribute("layout", layoutAttr);
 
-  // For now assume that the workload is at max 3D. So arguments to the region
-  // are workload along x, y and z.
   std::unique_ptr<Region> region;
   SmallVector<OpAsmParser::UnresolvedOperand, 4> regionOperands;
   SmallVector<Type, 4> regionTypes;
-  OptionalParseResult parseResult =
-      parser.parseOptionalRegion(region, regionOperands, regionTypes);
-  if (!parseResult.hasValue()) return success();
-  if (failed(*parseResult)) return failure();
+  // A missing optional region is materialized as an empty region.
+  (void)parser.parseOptionalRegion(region, regionOperands, regionTypes);
```

Comment:
Just use `parseRegion` ?

Reply:
I tried but it will complain with `error: expected '{' to begin a region` in various places.